src/directives/for.js

```diff
@@ -95,8 +95,12 @@ function evaluateItemsAndReturnEmptyIfXIfIsPresentAndFalseOnElement(component, e
     let items = component.evaluateReturnExpression(el, iteratorNames.items, extraVars)
 
     // This adds support for the `i in n` syntax.
-    if (isNumeric(items) && items > 0) {
-        items = Array.from(Array(items).keys(), i => i + 1)
+    if (isNumeric(items)) {
+        if(items > 0) {
+            items = Array.from(Array(items).keys(), i => i + 1)
+        } else if(items === 0) {
+            items = []
+        }
```

Comment:
You could probably turn this into a ternary instead, because the `else if` is irrelevant since we don't actually support negative numbers either.
```suggestion
    if (isNumeric(items)) {
        items = items > 0 ? Array.from(Array(items).keys(), i => i = 1) : []
    }
```